README.md

```diff
@@ -2,95 +2,75 @@
+[Discord](https://discord.gg/X7ystzGcf5) | [Website](https://evo.ninja) | Give the repo a :star: !  
 
-## Welcome to evo.ninja
+## Welcome!
 
-**The AI that evolves in real-time.** 
+To get started using evo.ninja simply head to our [website](https://evo.ninja), or to build and run from source follow these [setup instructions](#setup).
 
-It executes scripts to achieve a goal. It is capable of using fuzzy search to find and execute any script in its library. Scripts are namespaced JavaScript functions with typed arguments and a description. If it can not find a script, it will write one itself.
+## How it works
 
-Dive in to explore the capabilities and features provided by this agent.
-Before you can run evo.ninja, ensure you have Node.js and yarn installed.
+What makes evo.ninja special is that it evolves itself in real-time, based on the tasks at hand. Evo utilizes pre-defined agent personas that are tailored to specific domains of tasks. Each iteration of evo's execution loop it will select and adopts the persona that fits the task at hand best.
 
-## Examples
-- `Create one piece of SVG art and save it as art.svg`
-- `Divide 590 by 204 and save it to a file named output.txt`
-- `Create a CSV file named output.txt with the numbers from 1 to 10 and verify the content`
-- `Write the word Washington to the file called output.txt`
-- `Calculate the (590 * 204) + (1000 / 2) - 42`
-- `Fetch the price of ethereum, bitcoin and dogecoin and save them in a file named crypto.csv`
+### Agent Personas
+
+| Agent | Description |
+|-|-|
+| 📝[Synthesizer](./packages/agents/src/agents/Synthesizer/index.ts) | An assistant that handle files in workspace |
+| #️⃣ [Csv Analyst](./packages/agents/src/agents/CsvAnalyst/index.ts) | An expert analyzing and modifying CSV datasets. |
+| 🌐 [Researcher](./packages/agents/src/agents/Researcher/index.ts) | Advanced web information retriever that allows to do complex research in the internet. |
+| 💻 [Developer](./packages/agents/src/agents/Developer/index.ts) | A developer assistant that excels at coding related tasks with access to the file system. It plans and writes clean and effective code to files. |
+
+### Execution Loop
+
+1. **Predict Next Step:** For each iteration of the execution loop, Evo starts by making an informed prediction about what the best-next-step should be.
+2. **Select Best Agent:** Based on this prediction, Evo selects a best-fit agent persona.
+3. **Contextualize Chat History:** Based on the prediction from step 1, and the agent persona in step 2, the complete chat history is "contextualized" and only the most relevant messages are used for the final evaluation step.
+4. **Evaluate and Execute:** Using a smart LLM (GPT-4), a final evaluation step is run to determine what function call should be executed in order to achieve the user's goal. Once the function call has been executed, the response is stored, and the loop repeats itself until the user's goal has been accomplished.
```

Comment:
I think step 4. is to "implementation descriptive". Step 2 talks about persona selection without mentioning vector search. Step 1 talks about prediction without mentioning a RAG LLM call. I think we should do the same with 4.

Also Is saying "GPT-4" specifically necessary?